src/lib.rs

```diff
@@ -68,25 +80,39 @@ pub struct Args {
     /// Note that cargo generate won't overwrite an existing directory, even if `--force` is given.
     #[structopt(long, short)]
     pub force: bool,
+
     /// Enables more verbose output.
     #[structopt(long, short)]
     pub verbose: bool,
+
+    /// Use specific configuration file. Defaults to $CARGO_HOME/cargo-generate or $HOME/.cargo/cargo-generate
+    #[structopt(short, long, parse(from_os_str))]
+    pub config: Option<PathBuf>,
 }
 
-pub fn generate(args: Args) -> Result<()> {
+pub fn generate(mut args: Args) -> Result<()> {
+    if args.list_favorites {
+        return list_favorites(&args);
+    }
+    resolve_favorite(&mut args)?;
+
     let name = match args.name {
         Some(ref n) => ProjectName::new(n),
         None => ProjectName::new(interactive::name()?),
     };
 
     create_git(args, &name)?;
-
     Ok(())
 }
 
 fn create_git(args: Args, name: &ProjectName) -> Result<()> {
     let force = args.force;
-    let config = GitConfig::new_abbr(&args.git, args.branch.to_owned())?;
+    let config = GitConfig::new_abbr(
+        &args
```

Comment:
here too, what you find more often is this idiom:

```rust
        &args.git
            .context("Missing option git, or a favorite")?
```

Reply:
I'll fix this too